### PR TITLE
maid: init at 0.10

### PR DESCRIPTION
**Pull request by alinnow**
## Description of changes

Rule-based file operation tool.

https://github.com/maid/maid

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ShamrockLee — COMMENTED** [review]
Built and tested the binary functionality.

Some changes suggested. Other wise, LGTM.

**Review by ShamrockLee — APPROVED** [review]
Built and tested the binary functionality. LGTM.

**Review by ShamrockLee — COMMENTED** [review]
Thanks for writing the test case!

File: pkgs/by-name/ma/maid/test.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, maid }:
+
+stdenv.mkDerivation {
```
[the remark this answers is not in the document]
Comment (alanpearce) [reply pkgs/by-name/ma/maid/test.nix:3]: Mmm, I had a read-only filesystem in that case. Is there some way around that?

[the remark this answers is not in the document]
Comment (ShamrockLee) [reply pkgs/by-name/ma/maid/test.nix]: `runCommandLocal` internally calls `stdenvNoCC.mkDerivation` and specifies all the build commands with `buildCommand`.

What's wrong with it on a "read-only file system"? After all, you need a writable store to build packages.

[the remark this answers is not in the document]
Comment (ShamrockLee) [reply pkgs/by-name/ma/maid/test.nix]: `HOME` is intentionally set to an non-existing path `/homeless-shelter` inside the Nix build sandbox. It makes sense to give it a temporary home for testing.

File: pkgs/by-name/ma/maid/test.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{ runCommandLocal, maid }:
+
+runCommandLocal { nativeBuildInputs = [ maid ]; }
+  ''
```
Comment (ShamrockLee) [inline pkgs/by-name/ma/maid/test.nix:4]:
```suggestion
runCommandLocal "test-maid-run" {
  nativeBuildInputs = [ maid ];
} ''
```

The positional argument for package name is required when calling `runCommandLocal`.

**Review by ShamrockLee — APPROVED** [review]
LGTM!

I'm not familiar with Ruby packaging. Here are the reasons I approve this PR:
* both `maid` and `maid.passthru.tests.run` build
* the `package.nix` and `test.nix` looks sensible
* the bin functionality of the resulting derivation is good (tested according to the upstream README)

File: pkgs/by-name/ma/maid/package.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  gemdir = ./.;
+  exes = [ "maid" ];
+
+  passthru.updateScript = bundlerUpdateScript "maid";
+
+  passthru.tests.run = callPackage ./test.nix { };
+
+  meta = with lib; {
+    description = "Rule-based file mover and cleaner in Ruby";
+    homepage = "https://rubygems.org/gems/maid";
```
Comment (SuperSandro2000) [inline pkgs/by-name/ma/maid/package.nix:18]:
```suggestion
    homepage = "https://github.com/maid/maid";
```

File: pkgs/by-name/ma/maid/package.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ma/maid/package.nix:19]:
```suggestion
    license = licenses.gpl2Only;
```

**Pull request closed without merging**